### PR TITLE
[4] New moduleposition below-top, option to switch off logo

### PR DESCRIPTION
**Pull request by chmst**
### Summary of Changes
Not all users want it have a logo in the header of their site. Cassipoeia has no option to switch off the display of a logo or description
This PR adds a param to the template where the user can switch off the logo.
It adds a new position "below-top" where users can add an own module. 
Using both - position "below-top" and logo is up to the user. 

The error page can be adapted if this is accepted.

### Testing Instructions
Apply the patch and play with the template style cassiopeia. Use the module position brand 

### Actual result BEFORE applying this Pull Request
There is always an area for the logo above the navigation. 

### Expected result AFTER applying this Pull Request

The user can swith off the logo
grafik

### Documentation Changes Required
yes

### CODE REVIEW

File: templates/cassiopeia/templateDetails.xml [hunk 1]
```diff
@@ -47,11 +48,24 @@
 	<config>
 		<fields name="params">
 			<fieldset name="advanced">
+				<field
+					name="brand"
+					type="radio"
+					default="1"
+					label="TPL_CASSIOPEIA_BRAND_LABEL"
+					layout="joomla.form.field.radio.switcher"
+					filter="integer"
```
Comment (brianteeman) [inline templates/cassiopeia/templateDetails.xml]: shoudlnt this be filter boolean?

File: templates/cassiopeia/templateDetails.xml [hunk 2]
```diff
@@ -47,11 +48,24 @@
 	<config>
 		<fields name="params">
 			<fieldset name="advanced">
+				<field
+					name="brand"
+					type="radio"
+					default="1"
+					label="TPL_CASSIOPEIA_BRAND_LABEL"
```
Comment (brianteeman) [inline templates/cassiopeia/templateDetails.xml:53]: Please move label above default to match the codestyle rules

**Comment by brianteeman** [comment]
It is confusing having a module and an options with the same name that control different things.
The way it is used here in the options it is not clear what it does unless you read the code
Is it intentional that you can have two brands?

I think you need to rethink the approach here.

**Comment by chmst** [comment]
Have you a suggestion for a name of the module position? Maybe below-top?



<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33751">issues.joomla.org/tracker/joomla-cms/33751</a>.</sub>

**Comment by brianteeman** [comment]
Thinking out aloud.

How about instead of saying Brand Yes|No maybe Template|Module would make more sense?

**Comment by chmst** [comment]
I agree that the name brand for a moduleposition is wrong, as the position canbe used for everything. It is independent from 
displaying the brand ( = Logo / Description and Tagline ).

File: templates/cassiopeia/templateDetails.xml (hunk 2 above)
Comment (chmst) [reply templates/cassiopeia/templateDetails.xml:53]: done ,thx

File: templates/cassiopeia/templateDetails.xml [hunk 3]
```diff
@@ -47,11 +48,24 @@
...
+					filter="boolean"
+					>
+					<option value="0">JNO</option>
+					<option value="1">JYES</option>
+				</field>
+
 				<field
 					name="logoFile"
 					type="media"
 					default=""
```
Comment (hans2103) [inline templates/cassiopeia/templateDetails.xml:65]: I wonder... is `default=""` required or can it be removed?

File: templates/cassiopeia/index.php [hunk 4]
```diff
@@ -125,16 +125,25 @@
...
-		<div class="grid-child">
-			<div class="navbar-brand">
-				<a class="brand-logo" href="<?php echo $this->baseurl; ?>/">
-					<?php echo $logo; ?>
-				</a>
-				<?php if ($this->params->get('siteDescription')) : ?>
-					<div class="site-description"><?php echo htmlspecialchars($this->params->get('siteDescription')); ?></div>
-				<?php endif; ?>
+		<?php if ($this->countModules('below-top')) : ?>
+			<div class="grid-child container-brand">
```
Comment (hans2103) [inline templates/cassiopeia/index.php]: you'r counting for module position below-top and naming the inner container `container-brand`. That is confusing to me.

Comment (brianteeman) [reply templates/cassiopeia/index.php]: No difference to

https://github.com/joomla/joomla-cms/blob/8e3dcfba103a93cd1a7f0d40da3dce7fb9462825/templates/cassiopeia/index.php#L138-L151

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on dba8af8c3da93f8b1ae1c3bdafd2249120a8d220<br /><br />Similar position in github named "Header".<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33751">issues.joomla.org/tracker/joomla-cms/33751</a>.</sub>

File: templates/cassiopeia/index.php (hunk 4 above)
Comment (chmst) [reply templates/cassiopeia/index.php]: fixed

**Comment by chmst** [comment]
I have changed the name, thanks for the hint. 

The position below-top can be be used independent form the brand ( logo / description + tagliene).

grafik

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 7894923bb96ee66151e6830d3223dc554c595e74

#### 1. Setting in template style:

> Screen Shot 2021-05-12 at 13 05 24

#### 2. Frontend:

> Screen Shot 2021-05-12 at 13 08 49

#### 3. Frontend + Language switcher in position "below-top":

> Screen Shot 2021-05-12 at 13 09 23


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33751">issues.joomla.org/tracker/joomla-cms/33751</a>.</sub>

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 7894923bb96ee66151e6830d3223dc554c595e74<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33751">issues.joomla.org/tracker/joomla-cms/33751</a>.</sub>

**Comment by richard67** [comment]
@chmst Please add the missing language string for the new module position to file `language/en-GB/tpl_cassiopeia.sys.ini` so that people see a nice module position name in the pulldown, like we have it e.g. for the "Menu" position, where we see "Menu" and not "menu".

The name of the language string has to be: `TPL_CASSIOPEIA_POSITION_BELOW-TOP`.

**Comment by chmst** [comment]
Language string added, Thanks @richard67

**Comment by richard67** [comment]
> Language string added, Thanks @richard67

As far as I know, the name of the language string after the last underscore has to be equal to the name in the XML converted to uppercase.

That means it should be `TPL_CASSIOPEIA_POSITION_BELOW-TOP`.

But you have added `TPL_CASSIOPEIA_POSITION_BELOW_TOP`.

Just below you can see for bottom-a and bottom-b how it shoud be.

**Comment by richard67** [comment]
@sandramay0905 @ChristineWk Could you briefly test this PR again? It has received a change so that when you select the new module position for a module, you see the nice, translatable language string "Below Top" in the dropdown, instead of the raw "below-top" from the XML file. Just test this new change, the rest hasn't changed. Thanks in advance, and thanks so far for all previous testing here and elsewhere.

File: templates/cassiopeia/templateDetails.xml (hunk 3 above)
Comment (chmst) [reply templates/cassiopeia/templateDetails.xml:65]: It is a matter of taste. So let it be visible as before the patch.

**Comment by ChristineWk** [comment]
Should be OK:

screen shot 2021-05-12 at 20 33 34<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33751">issues.joomla.org/tracker/joomla-cms/33751</a>.</sub>

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 231cc0f3b9a59414ccf784e0ca09af8c3ea0f2f3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33751">issues.joomla.org/tracker/joomla-cms/33751</a>.</sub>

**Comment by richard67** [comment]
@ChristineWk What I meant with the changed thing to be tested was the name of the module position in backend, when you select the position:



Before the last change, it was just "below-top", or maybe "below-top [below-top]", now it has the "Below Top" at the beginning, which is a translatable text.

**Comment by ChristineWk** [comment]
Ah, OK :-)
screen shot 2021-05-12 at 20 51 20<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33751">issues.joomla.org/tracker/joomla-cms/33751</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 231cc0f3b9a59414ccf784e0ca09af8c3ea0f2f3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33751">issues.joomla.org/tracker/joomla-cms/33751</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on 231cc0f3b9a59414ccf784e0ca09af8c3ea0f2f3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33751">issues.joomla.org/tracker/joomla-cms/33751</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 231cc0f3b9a59414ccf784e0ca09af8c3ea0f2f3<br /><br />sorry this is a good test<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33751">issues.joomla.org/tracker/joomla-cms/33751</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33751">issues.joomla.org/tracker/joomla-cms/33751</a>.</sub>

**Comment by PhilETaylor** [comment]
I totally missed this PR. 

Is this the reason - after a clean install - I no longer get a header on the frontend... I thought I was going crazy for the last 15 mins! 

Screenshot 2021-05-14 at 19 44 00

**Comment by PhilETaylor** [comment]
I would say this PR is broken... because Brand = On in the admin console on a default fresh install - but I dont get a header, until I save the admin params.

**Comment by brianteeman** [comment]
Shouldnt the param have been set in the database on install? (it hasnt been and therefore the problem)

**Comment by richard67** [comment]
@brianteeman @PhilETaylor I'll make a PR soon.

**Comment by richard67** [comment]
@brianteeman @PhilETaylor PR #33881 is ready, please test. Thanks in advance.

**Comment by infograf768** [comment]
While I was working on https://github.com/joomla/joomla-cms/pull/31275/ I discovered that I had issues when using the lang switcher with flags as a dropdown in the new position.

switcher_top_position

I will have to add the color in the module css
```
div.mod-languages .btn-group .btn {
    flex: none;
    color: #000;
}
```

**Comment by richard67** [comment]
@brianteeman @PhilETaylor I've meanwhile replace my PR by a new, better one, for the issue with the header section not shown after a new installation or an update before having saved template style parameters. Please test #33930 . Thanks in advance.

**Pull request merged**